### PR TITLE
Customer Invoice - Add Master List Items - Exclude Items from combined MasterList that are not visible in the store

### LINKED ISSUE #783: Customer invoice > "Add master list items" adds items tablet store can't see
Build Number: 2.2.0-rc0 ( probably exists earlier )

Description: 
untitled

Here are a bunch of items in the customer invoice visible to the customer store, but not this store. So naturally there is no available stock

Reproducible: Yes

Reproduction Steps:
1. Create a customer invoice for a store with different item visibility than the store being worked on
2. See all the items that this store can't see and therefore has no available stock for

Comments:
Should only add the intersection of items between supplying store and customer name to the customer invoice I reckon.

### PR DESCRIPTION
**Pull request by sworup**
Fixes #783 

- Customer Invoice > Add Master List Items 
- Exclude Items from combined MasterList that are not visible in the store
- Done as per discussion with @ujwalSussol

### CODE REVIEW

**Comment by Chris-Petty** [comment]
I'm pretty sure that this approach will work (just filter what isn't visible in this store) but don't quite have a moment to test it today.

File: src/database/DataTypes/Transaction.js [hunk 1]
```diff
@@ -114,7 +114,9 @@ export class Transaction extends Realm.Object {
       this.otherParty.masterLists.forEach(masterList => {
         const itemsToAdd = complement(masterList.items, this.items, item => item.itemId);
         itemsToAdd.forEach(masterListItem => {
-          if (!masterListItem.item.crossReferenceItem) {
+          if (!masterListItem.item.crossReferenceItem
+            && database.objects('Item')
```
Comment (andreievg) [inline src/database/DataTypes/Transaction.js]: masterListItem.item.isVisible

Comment (sworup) [reply src/database/DataTypes/Transaction.js]: I did not realize masterlist had visible field. Will do this.

Comment (andreievg) [reply src/database/DataTypes/Transaction.js]: When you get a realm db object, you can access related records just like object

**Comment by sworup** [comment]
@andreievg Requisition code also has add from masterlist like transaction does
https://github.com/openmsupply/mobile/blob/a5c005a4cbb37dcda5cb8b863874048047f1628f/src/database/DataTypes/Requisition.js#L96-L101

**Comment by sworup** [comment]
> mobile/src/database/DataTypes/Requisition.js

If the Requisition checks the store's master lists then the items in the list should by default be visible to the store. Maybe we do not need to change this code, but leaving the comment so that someone could confirm.

**Pull request merged**